src/NuGet.Clients/VsExtension/NuGetPackage.cs

```diff
@@ -390,10 +390,10 @@ private IEnumerable<NuGet.Credentials.ICredentialProvider> GetCredentialProvider
                 VisualStudioAccountProvider.FactoryMethod,
                 errorDelegate);
 
-            var vstsProvider = importer.GetProvider();
-            if (vstsProvider != null)
+            var importedProviders = importer.GetProviders();
+            if (importedProviders.Count != 0)
             {
-                credentialProviders.Add(vstsProvider);
+                credentialProviders.AddRange(importedProviders);
             }
 
             var webProxy = (IVsWebProxy)GetService(typeof(SVsWebProxy));
```

Comment:
You may need to test that VSTS still works after the change, but if it doesn't then it means that ordering is extremely important here and a better system it needed for determining which providers get to handle a source.